frontend/src/hooks/useKpi.tsx

```diff
@@ -9,33 +9,41 @@ export default function useKpi() {
     const [kpis, setKpis] = useState<Kpi[]>([]);
 
     useEffect(() => {
-        getAllKpisAdmin()
+        getAllKpis()
     }, [])
 
-    const getAllKpisAdmin = () => {
+    const getAllKpis = () => {
         axios.get("/api/kpis")
             .then((response) => response.data)
             .then(setKpis)
     }
 
-    const addNewKpi = (name: string, targetForKpi: { targetValueOperator: string, targetValue: number, targetValueUnit: string }) => {
-        const newKpi: NewKpi = {name: name, targetForKpi: targetForKpi}
+    const addNewKpi = (newKpi: NewKpi) => {
         return axios.post("/api/kpis", newKpi)
             .then((response) => {
                     return response.data
                 }
             )
-            .then(getAllKpisAdmin);
+            .then(getAllKpis);
     }
 
     const deleteKpiById = (id: string) => {
-        return axios.delete("api/kpis/" + id)
+        return axios.delete(`/api/kpis/${id}`)
             .then((response) => response.status)
-            .then(getAllKpisAdmin)
+            .then(getAllKpis)
             .catch(() => {
                 toast.error("Die Kennzahl konnte nicht gefunden werden.")
```

Comment:
Du hast in Backend schon custom  Exception. Es wäre schön eine Nachricht von Backend hier zu schicken. Dann wenn in Zukunft du zwei verschieden Fehler "catchen" könntest, bekommst du immer richtige Nachricht.
Aber ja, nur wenn es für dich Sinn macht :-)